internal/resolvers/default/media_resolver.go

```diff
@@ -0,0 +1,133 @@
+package defaultresolver
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"mime"
+	"net/http"
+	"strings"
+
+	"github.com/Chatterino/api/pkg/humanize"
+	"github.com/Chatterino/api/pkg/resolver"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const mediaTooltipTemplateString = `<div style="text-align: left;">
+<b>Media File</b>
+{{if .MediaType}}<br><b>Type:</b> {{.MediaType}}{{end}}
+{{if .Size}}<br><b>Size:</b> {{.Size}}{{end}}
+</div>
+`
+
+var mediaTooltipTemplate = template.Must(template.New("mediaTooltipTemplate").Parse(mediaTooltipTemplateString))
+
+type mediaTooltipData struct {
+	MediaType string
+	Size      string
+}
+
+type MediaResolver struct {
+	baseURL string
+}
+
+func (r *MediaResolver) Check(ctx context.Context, contentType string) bool {
+	spl := strings.Split(contentType, "/")
+	switch spl[0] {
+	case "video", "audio":
+		return true
+	}
+	return false
+}
+
+func (r *MediaResolver) Run(ctx context.Context, req *http.Request, resp *http.Response) (*resolver.Response, error) {
+	mimeType := resp.Header.Get("Content-Type")
+	spl := strings.Split(mimeType, "/")
+
+	size := ""
+	reportedSize := resp.ContentLength
+	if reportedSize > 0 {
+		size = humanize.Bytes(uint64(reportedSize))
+	}
+
+	ttData := mediaTooltipData{
+		MediaType: fmt.Sprintf("%s (%s)",
+			cases.Title(language.English).String(spl[0]),
+			strings.ToUpper(extensionFromMime(mimeType)),
+		),
+		Size: size,
+	}
+
+	var tooltip bytes.Buffer
+	if err := mediaTooltipTemplate.Execute(&tooltip, ttData); err != nil {
+		return nil, err
+	}
+
+	targetURL := resp.Request.URL.String()
+	response := &resolver.Response{
+		Status:  http.StatusOK,
+		Link:    targetURL,
+		Tooltip: tooltip.String(),
```

Comment:
I removed the PathEscape here and it still works fine in Chatterino. Is there any actual need for it?